src-cpp/rdkafkacpp.h

```diff
@@ -3489,4 +3518,47 @@ class Metadata {
 
 }
 
+/**
+ * @name Memory allocator
+ * @{
+ *
+ */
+
+/**
+ * @brief Allocate memory using the same allocator librdkafka uses.
+ *
+ * This is typically an abstraction for the malloc(3) call and makes sure
+ * the application can use the same memory allocator as librdkafka for
+ * allocating pointers that are used by librdkafka.
+ *
+ * \p handle of type RdKafka::Handle can be set to return memory allocated by a
+ * specific \c handle instance otherwise pass NULL for \p handle.
+ *
+ * @remark Memory allocated by mem_malloc() must be freed using
+ *         mem_free().
+ */
+RD_EXPORT
+void *mem_malloc (RdKafka::Handle *handle, size_t size);
```

Comment:
I was thinking that these two are put as instance methods on the Handle class so that the Handle argument is not needed.

Reply:
That still wouldn't solve the NULL handle case. Since for example Headers has no handle.

Reply:
Or should I keep the void *RdKafka::mem_malloc (size_t size); version around for just these cases?

Reply:
Also my guess is that if I'll include it in Handle, then you want them pure virtual